src/test/java/htsjdk/samtools/cram/encoding/external/ExternalCodecEquivalenceTest.java

```diff
@@ -0,0 +1,224 @@
+package htsjdk.samtools.cram.encoding.external;
+
+import htsjdk.HtsjdkTest;
+import htsjdk.samtools.cram.encoding.CRAMCodec;
+import htsjdk.samtools.cram.io.IOTestCases;
+import htsjdk.samtools.util.RuntimeIOException;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ExternalCodecEquivalenceTest extends HtsjdkTest {
+    // show that External codecs are equivalent within certain ranges
+
+    // if all values are 0 <= v <= 0x7F (high bit 7)
+    // ExternalByteCodec, ExternalIntegerCodec, and ExternalLongCodec streams are identical
+    // because raw bytes, ITF8 and LTF8 are identical
+
+    @Test(dataProvider = "testPositiveByteLists", dataProviderClass = IOTestCases.class)
+    public void byteEquivalenceTest(final List<Byte> values) {
+        codecPairTestForByte(values, this::writeByteForByte, this::readIntegerForByte);
```

Comment:
I'd like to make this into a DataProvider somehow, but that doesn't appear to be possible with functional interfaces

Reply:
Yeah, unfortunately the `FunctionalInterface`->`Object` conversion doesn't usually go well.